### PR TITLE
Improves loading of Push Adapter, fix loading of S3Adapter

### LINKED ISSUE #820: Upgrade to 2.1.4 breaks S3 fileAdapter
I just upgraded my server to 2.1.4 and my fileAdapter seems to have broken.  The files are now being pointed to my own server endpoint instead of S3. 

My server config looks like:

```
  filesAdapter: new S3Adapter(
    process.env.AWS_ACCESS_KEY,
    process.env.AWS_SECRET_ACCESS_KEY,
    'whiterabbitapps',
    {directAccess: true}
  ),
```

I removed the fileKey config just in case that was interfering and that didn't fix the problem.  Any insight into why this may have broken?

### LINKED ISSUE #832: Parse S3 File Access with No Credentials
Make sure these boxes are checked before submitting your issue -- thanks for reporting issues back to Parse Server!

-[X] You've met the [prerequisites](https://github.com/ParsePlatform/parse-server/wiki/Parse-Server-Guide#prerequisites).

-[X] You're running the [latest version](https://github.com/ParsePlatform/parse-server/releases) of Parse Server.

-[X] You've searched through [existing issues](https://github.com/ParsePlatform/parse-server/issues?utf8=%E2%9C%93&q=). Chances are that your issue has been reported or resolved before.

#### Environment Setup
AWS, Parse 2.1.4, using S3 File Adapter

#### Steps to reproduce
Upload file through parse.  Url is returned.

Hit new file through Parse (directAccess: false) with no credentials and it returns properly.  I believe in previous versions this request required Parse credentials.  Is this by design?

### PR DESCRIPTION
**Pull request by flovilmart**
- Adds environment variables to configure S3Adapter

Fixes #820
Fixes #832

### CODE REVIEW

**Comment by drew-gross** [comment]
Can you also update the docs to describe the new options?

**Comment by flovilmart** [comment]
Sure!

File: src/Adapters/AdapterLoader.js [hunk 1]
```diff
@@ -28,13 +28,6 @@ export function loadAdapter(adapter, defaultAdapter, options) {
...
     return loadAdapter(adapter.adapter, undefined, adapter.options);
-  } else {
-    // Try to load the defaultAdapter with the options
-    // The default adapter should throw if the options are
-    // incompatible
-    try {
-      return loadAdapter(defaultAdapter, undefined, adapter);
-    } catch (e) {};
   }
   // return the adapter as is as it's unusable otherwise
```
Comment (mcdonamp) [inline src/Adapters/AdapterLoader.js]: Is this comment still correct, or will it just return the object representing the adapter (which is presumably usable)?

Comment (flovilmart) [inline src/Adapters/AdapterLoader.js]: Yeah? That retuns the adapter as not other loading method seems to be available.

Comment (mcdonamp) [inline src/Adapters/AdapterLoader.js]: Oh man, I totally misread that, somehow turned that into "return the adapter as it's unusable". Might I suggest "return the adapter as provided" or something similar?

Comment (flovilmart) [inline src/Adapters/AdapterLoader.js]: :+1:

**Pull request merged**